hyperapp.d.ts

```diff
@@ -54,56 +56,58 @@ export type ActionResult<State extends Hyperapp.State> =
   | null
   | void
 
-export type ActionCallers<
-  State extends Hyperapp.State,
-  Actions extends Hyperapp.Actions<State>
-> = {
-  [P in keyof Actions]: (
-    data: any
-  ) => ActionResult<State> | ActionCallers<State, Actions>
-}
 
-export interface Actions<State extends Hyperapp.State> {
+export interface Actions {
   [action: string]:
-    | Actions<State>
+    | Actions
     | ((
-        state: State,
-        actions: ActionCallers<State, Actions<State>>,
         data: any
       ) => ActionResult<State>)
 }
 
+export type ActionsImpl<
+State extends Hyperapp.State,
+Actions extends Hyperapp.Actions
+> = {
+[P in keyof Actions]: (
+  state: State,
+  actions: Actions,
+  data: any
+) => ActionResult<State> | ActionsImpl<State, Actions>
```

Comment:
Can you add a test in index.tsx for action namespaces? I think that this type as it stands says that "`ActionImpl` is an object whose values are functions that return an `ActionResult`. Or the function can return an `ActionsImpl`" as opposed to what I think you want to say: "`ActionImpl` is an object whose values are either a function that returns an `ActionResult`. Or the object value can be an `ActionsImpl`"

Reply:
I think this bug existed previously but was on the `ActionCallers` interface

Reply:
You are right. Missing parentheses. The `|` is hard to play with ^^

Reply:
Hmm nested actions seem to be a little tricky. I think the body of `ActionsImpl` needs to include the parentheses as well as a Partial:
```ts
[P in keyof Actions]: ((
  state: State,
  actions: Actions,
  data: any
) => ActionResult<State>) | Partial<ActionsImpl<State, Actions>>
```
This fix would allow nested actions I think at least.